extract.py

```diff
@@ -0,0 +1,408 @@
+            #pickle.dump(files, open("config/files.pic", "wb"))
+            self.files = files
+        else: self.files = pickle.load(open("config/files.pic", "rb"))
+
+        f = pyfits.open(self.files[0])
+
+        self.flat = obs_par['flat']
+        self.LTV1 = int(f[1].header['LTV1'])
+        self.LTV2 = int(f[1].header['LTV2'])
+
+        self.POSTARG1 = int(f[0].header['POSTARG1'])
+        self.POSTARG2 = int(f[0].header['POSTARG2'])
+        self.platescale = 0.13
+
+        self.subarray_size = int(f[1].header['SIZAXIS1'])                #size of subarray
+        self.BEAMA_i = int(obs_par['BEAMA_i'])                    #start of first order trace
+        self.BEAMA_f = int(obs_par['BEAMA_f'])                    #end of first order trace
+        self.npix = self.BEAMA_f - self.BEAMA_i                    #length of trace
+        self.grism = obs_par['GRISM']
+
+
+        self.plot_trace = convert_to_bool(obs_par['plot_trace'])
+        self.diagnostics = convert_to_bool(obs_par['diagnostics'])
+        self.output = convert_to_bool(obs_par['output'])
+        if self.output == False: print("NOTE: output is set to False!")
+
+        self.sig_cut = float(obs_par['sig_cut'])
+        self.nsmooth = int(obs_par['nsmooth'])
+        self.window = int(obs_par['window'])                    #window outside of which the background is masked
+        self.nvisit = int(obs_par['nvisit'])
+        self.norb = int(obs_par['norb'])
+
+        self.t0 = float(fit_par['t0'])
+        self.period = float(fit_par['per'])
+
+        self.one_di_per_visit = convert_to_bool(obs_par['one_di_per_visit'])
+        if self.one_di_per_visit==True: norb = nvisit
+
+        self.ra = f[0].header['ra_targ']*math.pi/180.0                #stores right ascension
+        self.dec = f[0].header['dec_targ']*math.pi/180.0            #stores declination
+
+        self.expstart = f[0].header['expstart']                    #exposure start time
+        self.exptime = f[0].header['exptime']                    #exposure time [seconds]
+
+        idx = np.argsort(refpix[:,0])
+        self.refpix = refpix[idx]                        #reference pixels from direct image
+
+        self.torbstart = self.refpix[:,0]                    #start times for each orbit
+        self.torbstart = np.append(self.torbstart, 1.0e10)            #appends large value to make get_orbnum routine work
+
+        self.tstart = tstart
+
+        self.visnum = None
+        self.orbnum = None
+
+        self.wavegrid = None
+
+        self.coordtable = []                            #table of spacecraft coordinates
+        for i in range(self.nvisit): self.coordtable.append("bjd_conversion/horizons_results_v"+str(i)+".txt")
+
+
+def make_dict(table):
+    return {x['parameter']: x['value'] for x in table}
+
+def convert_to_bool(str):
+    if str=="True": return True
+    elif str=="False": return False
+    else: return "String not equal to True or False"
```

Comment:
There is a [built-in way to do this](https://stackoverflow.com/questions/53287229/python-parsing-true-and-false-as-strings):
```
In [31]: import ast

In [32]: ast.literal_eval('True')
Out[32]: True

In [33]: ast.literal_eval('False')
Out[33]: False
```
It returns a value error for other strings so easy to catch. Using this instead makes outputs unpredictable - either a boolean or a string. If the string is returned, the code below will crash. This goes back to better configuration.